### PR TITLE
[Doc] Remove duplicate pages

### PR DESCRIPTION
**Pull request by candlerb**
There are stale/overlapping pages under `site2/docs`:

* functions-debug.md
* functions-debugging.md  \<\< obsolete, not linked

and

* functions-deploy.md
* functions-deploying.md  \<\< obsolete, linked only from deploy-bare-metal

This patch removes the obsolete pages, and updates the links in deploy-bare-metal (also removing a link to functions-state.md which is also gone)

### CODE REVIEW

**Comment by sijie** [comment]
@tuteng @Jennifer88huang : since you are fixing the links between different versions of documentation, can you please take a look at this pull request?

**Comment by tuteng** [comment]
> @tuteng @Jennifer88huang : since you are fixing the links between different versions of documentation, can you please take a look at this pull request?

I will try my best to fix this problem this afternoon and then create a pr.

File: site2/docs/deploy-bare-metal.md [hunk 1]
```diff
@@ -447,7 +447,7 @@ bin/pulsar-admin functions create \
   --name exclamation
 ```
 
-Check if the function runs as expected by [triggering](functions-deploying.md#triggering-pulsar-functions) the function.
+Check if the function runs as expected by [triggering](functions-deploy.md#triggering-pulsar-functions) the function.
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md]:
```suggestion
Check if the function runs as expected by [triggering](functions-deploy.md#trigger-pulsar-functions) the function.
```

**Comment by Jennifer88huang** [comment]
@candlerb Since there are more obsolete files, @tuteng submit a PR to fix it in https://github.com/apache/pulsar/pull/5700 , could you please also check #5700 ? His solution includes the files in this PR.

**Comment by candlerb** [comment]
I don't understand #5700 at all; it says it's being done because of limitations of docusaurus, which is a platform I don't know.

**Comment by Jennifer88huang** [comment]
@candlerb Never mind. Let me explain in details.
Pulsar website use docusaurus as framework. 
Previously, we've tried to remove all those obsolete files, and it resulted to link failure for versions earlier than 2.4.1. And then @tuteng added all those obsolete files back to the `docs` folder. There was no link issues for the current version and earlier versions. However, contributors felt confused why those obsolete files are there in the master, and not sure which file to update. 

We're trying to remove all those obsolete files, yet it would inevitably affect links in earlier versions. Those obsolete files were used in earlier versions, and we have to maintain doc content in the earlier versions unchanged, and keep only latest doc content in the master. So @tuteng comes with the solution: keep the obsolete files name in the master(with empty content), move the obsolete files and contents to earlier versions.

If you have any better idea on this, feel free to contact us.

**Comment by tuteng** [comment]
> I don't understand #5700 at all; it says it's being done because of limitations of docusaurus, which is a platform I don't know.

Docusaurus is a framework with version management of documents https://docusaurus.io/docs/en/installation.
The sub-version depends on the main version. Observe the following `id` and `original_id` fields. If the main version file is deleted, the file under the subversion will not find the index. Users get a lot of Http 404 when viewing older versions of documents. However, if we don't delete the files and keep the contents under the main version, there will be many duplicate contents, which will lead to confusion when the developers contribute. therefore, we delete the contents and only keep one header in order for the sub-version to have a correct index, the contributor only needs to update the contents in the main version.
@candlerb 

`functions-api.md`
main version (latest version)
```
---
id: functions-api
title: The Pulsar Functions API
sidebar_label: API
---
```
Subversion, (old version)
```
---
id: version-2.4.0-functions-api
title: The Pulsar Functions API
sidebar_label: API
original_id: functions-api
---
```

**Comment by Jennifer88huang** [comment]
@candlerb could you please update the suggestion in the `deploy-bare-metal.md` file, and resolve conflicting files, and then we can go on with the next step? Thank you.

**Comment by candlerb** [comment]
I withdraw this PR.

**Pull request closed without merging**